core/environment.ts

```diff
@@ -170,28 +188,36 @@ export class Environment {
       }
     }
 
-    const constructor = new Function(
-      "__env",
-      `return${sync ? "" : " async"} function __template (${dataVarname}) {
-        let __pos = 0;
```

Comment:
Dropping `__pos` is not necessarily per se, but in order to pinpoint the location of a JS syntax error, the source code needs to be "read" anyway to figure out where the last tag started. To do that, we need to match `__pos = \d+;`, and so I figured we might as well remove the runtime variable and go with `/*__pos:\d+*/` instead.

JS syntax errors are, by the way, the reason I'm making this PR against `v2` and not `main`; the transformer changes the code in unpredictable ways, so you can no longer rely on JS from the template to occur as-is within the compiled template function. That, in turn, prevents us from showing the location of a JS syntax error within the original Vento source file.

Reply:
Whoops, I did see this when rebasing, then forgot 😬 will revert the overwritten bits!